### ISSUE TITLE
Add support to render moves with own numbers in --first-move-number --move-numbers mode

### ISSUE THREAD

**Issue by dany52** [julianandrews/sgf-render#24]
Now, `--first-move-number 51 --move-numbers 100` produce 1,2,...,50 numbers in the output

I need 51,52,...,100 numbers in the output

P.S. `sgf-render` near perfect tool! after some minor improvements, it has the potential to become an almost ideal solution for creating diagrams

**Comment by julianandrews** [comment]
This makes sense.

I could add a new `--starting-move-number` flag or something like that, but I think that would be pretty confusing with the existing `--first-move-number` flag.

It's a little verbose, but I think what I'm leaning towards is

- drop the `--first-move-number` flag,
- add a `--move-numbers-first-node` flag (default 0), and 
- add a `--move-numbers-first-number` flag (default 1).

The `--move-numbers-first-node` flag will behave like `--first-move-number` does now, and `--move-numbers-first-number` will let you set the number to start counting from.

Would this work for your use case?

**Comment by dany52** [comment]
>     * add a `--move-numbers-first-number` flag (default 1).
imho the `sgftopng` approach with `-from` option with default own number is the best:

```
./sgftopng  -coord           -var1 foo_var1_110-125.png 110-125 < foo.sgf        
./sgftopng  -coord           -var2 foo_var2_110-125.png 110-125 < foo.sgf        
./sgftopng  -coord  -from  1 -var1 foo_var1_from_1_110-125.png 110-125 < foo.sgf 
./sgftopng  -coord  -from 10 -var1 foo_var1_from_10_110-125.png 110-125 < foo.sgf
```

produce...

![ds_var1_110-125](https://user-images.githubusercontent.com/436366/160290266-20acbe47-f254-4d05-afb2-3aeeea1d4728.png)
![ds_var1_from_1_110-125](https://user-images.githubusercontent.com/436366/160290268-c87d5b98-77a9-4175-81cc-cce5f60a504c.png)
![ds_var1_from_10_110-125](https://user-images.githubusercontent.com/436366/160290270-49b1b5a4-c964-4ec2-aa71-0aa2a2c3f634.png)
![ds_var2_110-125](https://user-images.githubusercontent.com/436366/160290271-dc99c8a3-898a-4d9d-9c67-1749b1c1260e.png)

**Comment by julianandrews** [comment]
Thanks. That's helpful!

I don't want to follow `sgftopng` exactly since the default in `sgfrender` is no move numbers, but it's definitely better to allow specifying a range of moves (instead of just a starting move). So new plan:

- Change `--move-numbers` argument to accept an optional range of nodes to number (e.g. `--move-numbers 100-125`). If the argument is omitted, number all moves. If the end of the range is omitted, it will number all moves from the start onwards.
- Add a `--move-numbers-from` argument that specifies the number to start at (defaults to the actual move number of the first move).

So this is pretty similar to `sgftopng`,  just with slightly more verbose flag names, and a default of no move numbers (which I think I'd like to preserve).

Any thoughts?

**Comment by dany52** [comment]
Yes, this is similar to ```sgftopng``` But we still have ```sgfrender``` headache to point to variations moves. Especially in compared to ```sgftopng``` clear way via ```-var & range moves``` options

**Comment by julianandrews** [comment]
> Yes, this is similar to `sgftopng` But we still have `sgfrender` headache to point to variations moves. Especially in compared to `sgftopng` clear way via `-var & range moves` options

Ok, sounds like a plan! 

Taking a look at how `sgftopng` handles selecting variations, I've got to agree it's a lot easier to work with. I've opened a new issue (#25) to update how node selection works. I should be able to make the move number changes pretty quickly. Node selection may take a little longer since I'll want to spend some time thinking about the design, and will need to implement some sort of variation info display.
